sycl/test-e2e/Matrix/joint_matrix_rowmajorA_rowmajorB_impl.hpp

```diff
@@ -108,42 +107,43 @@ int main() {
       q.get_device()
           .get_info<sycl::ext::oneapi::experimental::info::device::
                         matrix_combinations>();
-  for (unsigned int i = 0; i < combinations.size(); i++) {
-    if (combinations[i].atype == matrix_type::bf16) {
-      if (combinations[i].nsize == 0 ||
-          (combinations[i].nsize == 16 && combinations[i].max_msize == 8 &&
-           combinations[i].ksize == 16)) {
-        gemm_row_major<16, 16, class gemm_bfloat16_16, bfloat16, bfloat16,
-                       float>();
-      }
-      if (combinations[i].nsize == 8 && combinations[i].max_msize == 8 &&
-          combinations[i].ksize == 16) {
-        gemm_row_major<8, 16, class gemm_bfloat16_8, bfloat16, bfloat16,
-                       float>();
+  int res = 0;
+  for (auto &combination : combinations) {
+    if (combination.nsize == 0 ||
+        combination.nsize == 16) { // Intel AMX or architecture::intel_gpu_pvc
+      res += gemm_row_major<8, 16, 16, class bf16_8x16x16, bfloat16, bfloat16,
+                            float>();
+      res += gemm_row_major<8, 16, 32, class ss_8x16x32, int8_t, int8_t,
+                            int32_t>();
+      res += gemm_row_major<8, 16, 32, class us_8x16x32, uint8_t, int8_t,
+                            int32_t>();
+      res += gemm_row_major<8, 16, 32, class su_8x16x32, int8_t, uint8_t,
+                            int32_t>();
+      res += gemm_row_major<8, 16, 32, class uu_8x16x32, uint8_t, uint8_t,
+                            int32_t>();
+
+      if (combination.nsize == 16) { // architecture::intel_gpu_pvc
+        res += gemm_row_major<1, 64, 16, class bf16_1x64x16, bfloat16, bfloat16,
+                              float>();
+        res += gemm_row_major<32, 64, 16, class bf16_32x64x16, bfloat16,
```

Comment:
Should we add one for 16x16x16 ?

Reply:
I thought about it, but then I realized that matrix B would be the same as for 8x16x16 combination and decided not to add it, so it doesn't look like we would test more by adding this combination.
At the same time we test here 1x64x16 and 32x64x16, while matrix B is the same for these 2 combinations as well.
So I decided not to spend time on adding 8x16x16 or removing one of 1x64x16 or 32x64x16.
But if you have strong opinion on adding 16x16x16 or removing one of 1x64x16 or 32x64x16 I can do that...